lib/ProductOpener/Products.pm

```diff
@@ -289,11 +290,49 @@ sub normalize_code ($code) {
 
 	if (defined $code) {
 		($code, my $gs1_ai_data_str) = &normalize_code_with_gs1_ai($code);
+		$code = normalize_code_zeroes($code);
 	}
 	return $code;
 }
 
-=head2 normalize_code_with_gs1_ai()
+=head2 normalize_code_zeroes($code)
+
+On disk, we store product files and images in directories named after the product code, and we add leading 0s to the paths.
+So we need to normalize the number of leading 0s of product codes, so that we don't have 2 products for codes that differ only by leading 0s.
+
+This function normalizes the product code by:
+- removing leading zeroes,
+- adding leading zeroes to have at least 13 digits,
+- removing leading zeroes for EAN8s to keep only 8 digits
```

Comment:
I don't understand why we keep this rule ?

Reply:
it's to keep barcodes closer to how they are written on packages. while it's possible to refer to a ean9 with 5 leading 0s, on the package, the 0s are not there (that's whole point of ean8).

as you noted, that way we won't change barcodes for all EAN8s